cmake/compilers/Clang.cmake

```diff
@@ -39,6 +47,9 @@ if (NOT APPLE)
     set(TBB_WARNING_SUPPRESS -Wno-parentheses)
 endif()
 
+if (NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 12)
+    set(TBB_LIB_COMPILE_FLAGS -mwaitpkg)
```

Comment:
`-mwaitpkg` is [x86](https://clang.llvm.org/docs/ClangCommandLineReference.html#x86) option, so it should be applied only for this architecture, consider moving it under `if (CMAKE_SYSTEM_PROCESSOR STREQUAL x86_64)` on line 33.

Consider adding the same key to GNU.cmake.